### PR TITLE
runtime: Support virtiofs queue size for qemu and make it configurable

### PR DESCRIPTION
**Pull request by darfux**
The default vhost-user-fs queue-size of qemu is 128 now. Set it to 1024 by default which is same as clh. Also make this value configurable.

Fixes: #5694

Signed-off-by: liyuxuan.darfux <liyuxuan.darfux@bytedance.com>

### CODE REVIEW

File: src/runtime/Makefile [hunk 1]
```diff
@@ -193,6 +193,7 @@ DEFVALIDVIRTIOFSDAEMONPATHS := [\"$(DEFVIRTIOFSDAEMON)\"]
 #if value is 0, DAX is not enabled
 DEFVIRTIOFSCACHESIZE ?= 0
 DEFVIRTIOFSCACHE ?= auto
+DEFVIRTIOFSQUEUESIZE ?= 1024
```
Comment (liubin) [inline src/runtime/Makefile:196]: Is 1024 an appropriate value?

File: src/runtime/pkg/govmm/qemu/qemu.go [hunk 2]
```diff
@@ -1503,6 +1504,11 @@ func (vhostuserDev VhostUserDevice) QemuFSParams(config *Config) []string {
 	deviceParams = append(deviceParams, driver)
 	deviceParams = append(deviceParams, fmt.Sprintf("chardev=%s", vhostuserDev.CharDevID))
 	deviceParams = append(deviceParams, fmt.Sprintf("tag=%s", vhostuserDev.Tag))
+	queueSize := uint32(1024)
+	if vhostuserDev.QueueSize != 0 {
```
Comment (liubin) [inline src/runtime/pkg/govmm/qemu/qemu.go:1508]: Does this keep the minimum change(Not sure setting a default 1024 does matter )?

```golang
if vhostuserDev.QueueSize != 0 {
	deviceParams = append(deviceParams, fmt.Sprintf("queue-size=%d", vhostuserDev.QueueSize))
}
```

File: src/runtime/Makefile (hunk 1 above)
Comment (darfux) [reply src/runtime/Makefile:196]: CLH uses 1024 now： https://github.com/kata-containers/kata-containers/blob/3bb145c63ad1fe8c335d28029ee8ad8ca5f4f340/src/runtime/virtcontainers/clh.go#L1589

File: src/runtime/pkg/govmm/qemu/qemu.go (hunk 2 above)
Comment (darfux) [reply src/runtime/pkg/govmm/qemu/qemu.go:1508]: As comment above, here uses 1024 to keep same with CLH. Qemu's default 128 is a bit small as FUSE uses a 256 pages limit:
https://github.com/torvalds/linux/blob/v6.0/fs/fuse/fuse_i.h#L39

**Review by liubin — APPROVED** [review]

File: src/runtime/Makefile (hunk 1 above)
Comment (fidencio) [reply src/runtime/Makefile:196]: 1024 is the default value, AFAIR.

**Review by fidencio — APPROVED** [review]
lgtm, thanks @darfux!

**Comment by fidencio** [comment]
@BbolroC, s390x main is failing, mind to check whether this could be related to this PR?

**Comment by BbolroC** [comment]
> @BbolroC, s390x main is failing, mind to check whether this could be related to this PR?

@fidencio ah, sorry for the late check. The error has nothing to do with this PR. It is a flaky one due to the cni configuration. 

I will fix this after the current task is finished. Thanks for the notification.

**Pull request merged**